src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -320,6 +320,12 @@ export class DataSourceSavedObjectsClientWrapper {
             '"auth.credentials.region" attribute is required'
           );
         }
+
+        if (!service) {
```

Comment:
Is multiple datasources out of experimental?

If so, then it will likely need a migration for this saved object otherwise previous saved objects will fail.

Reply:
Yes, MD is out of experimental.

I am removing this condition, because 1)  we are using dynamic mapping for data source saved object. 2) service name is actually defined as optional field. It can be undefined and won't break backwards compatibility

Reply:
This condition is added to the create API of saved object wrapper, to align with UI dropdown list, that a service name must be selected when creating the corresponding data source